build.sbt

```diff
@@ -869,6 +881,88 @@ def genSmithy(config: Configuration) = Def.settings(
 def genSmithyScala(config: Configuration) = genSmithyImpl(config).map(_._1)
 def genSmithyResources(config: Configuration) = genSmithyImpl(config).map(_._2)
 
+// SBT setting to specify artifacts to be included in the Smithy model for compliance testing
+val complianceTestDependencies =
+  SettingKey[Seq[ModuleID]]("complianceTestDependencies")
+
+// writes out a json representation of the smithy model pulled from Smithy4s dependencies config
+// result is cached using the dependency list as the cache key
+def dumpModel(config: Configuration): Def.Initialize[Task[Seq[File]]] =
+  Def.task {
+    val dumpModelCp = (`codegen-cli`.jvm(
+      Smithy4sBuildPlugin.Scala213
+    ) / Compile / fullClasspath).value
+      .map(_.data)
+    val mc = "smithy4s.codegen.cli.Main"
```

Comment:
you can probably use the `mainClass` task

```
sbt:sauce> inspect mainClass
[info] Task: scala.Option[java.lang.String]
[info] Description:
[info]  Defines the main class for packaging or running.
```

Reply:
whats the difference?

Reply:
one is hardcoded and the other one is found by the build tool

the latter is more _future proof_ (if we ever change the name)

Reply:
I like the idea of `mainClass` here

Reply:
great